db/models/models.js

```diff
@@ -7,11 +9,30 @@ exports.fetchAllTopics = () => {
   });
 };
 
-exports.fetchArticles = () => {
-  return db.query(`SELECT articles.* , COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments on comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY articles.created_at DESC;`).then((results) => {
-    return results.rows;
-  });
-};
+exports.fetchArticles = async (sortby = 'created_at', order = 'DESC', topics) => {
+  if (order !== 'DESC' && order !== 'ASC'){
+    return Promise.reject({ status: 404, msg: 'order not valid - must be ASC or DESC' });
+    }
+  if (!articles[0].hasOwnProperty(sortby)){
```

Comment:
Where will `articles[0]` be coming from? 🤔 

For this specific project, I would recommend just hardcoding valid sort options and check that the passed in `sortby` variable exists in the options.